src/models/ComponentStyle.js

```diff
@@ -78,12 +78,14 @@ export default (
      * */
     generateAndInjectStyles(executionContext: Object, styleSheet: StyleSheet) {
       const { isStatic, componentId, lastClassName } = this
-      if (areStylesCacheable && isStatic && lastClassName !== undefined) {
-        return lastClassName
-      }
-
```

Comment:
Here you're leading the code into expensive operations that we're trying to avoid